gpu/gpu.go

```diff
@@ -62,6 +62,7 @@ func initGPUHandles() {
 
 	// TODO - if the ollama build is CPU only, don't do these checks as they're irrelevant and confusing
 
+	gpuHandles = &handles{nil, nil}
```

Comment:
I'm a little confused how this is making any difference.

Reply:
Here's what appears to be occurring: when the preceding `case` statement is executed without match before `gpuHandles` is declared to conform to the struct, it returns only a nil value which triggers the error when testing for `gpuHandles.cuda` or `gpuHandles.rocm` elsewhere. Note I'm on: `go version go1.21.6 android/arm64`